code/modules/antagonists/traitor/contractor/datums/syndicate_contract.dm

```diff
@@ -0,0 +1,518 @@
+
+	var/outcome_text = ""
+	if(generate())
+		status = CONTRACT_STATUS_INACTIVE
+		outcome_text = "Luckily, there is another target on station we can interrogate. A new contract can be found in your uplink."
+	else
+		// Too bad.
+		status = CONTRACT_STATUS_INVALID
+		outcome_text = "Unfortunately, we could not find another target to interrogate and thus we cannot give you another contract."
+
+	if(owning_hub.uplink)
+		owning_hub.uplink.message_holder("[pre_text] [outcome_text]", 'sound/machines/terminal_prompt_deny.ogg')
+		SStgui.update_uis(owning_hub.uplink)
+
+/**
+  * Marks the contract as failed and stops it.
+  *
+  * Arguments:
+  * * difficulty - The visual reason as to why the contract failed.
+  */
+/datum/syndicate_contract/proc/fail(reason = "")
+	if(status != CONTRACT_STATUS_ACTIVE)
+		return
+
+	// Update info
+	owning_hub.current_contract = null
+	status = CONTRACT_STATUS_FAILED
+	fail_reason = reason
+	// Notify
+	clean_up()
+	owning_hub.uplink?.message_holder("You failed to kidnap the target, agent. Do not disappoint us again.", 'sound/machines/terminal_prompt_deny.ogg')
+
+/**
+  * Initiates the extraction process if conditions are met.
+  *
+  * Arguments:
+  * * M - The contractor.
+  */
+/datum/syndicate_contract/proc/start_extraction_process(obj/item/contractor_uplink/U, mob/living/carbon/human/M)
+	if(!U?.Adjacent(M))
+		return "Where in space is your uplink?!"
+	else if(status != CONTRACT_STATUS_ACTIVE)
+		return "This contract is not active."
+	else if(extraction_deadline > world.time)
+		return "Another extraction attempt cannot be made yet."
+
+	var/mob/target = contract.target.current
+	if(!target)
+		invalidate()
+		return "The target is no longer on our sensors. Your contract will be invalidated and replaced with another one."
+	else if(!contract.can_start_extraction_process(M, target))
+		return "You and the target must be standing in the extraction area to start the extraction process."
+
+	M.visible_message("<span class='notice'>[M] starts entering a cryptic series of characters on [U].</span>",\
+					  "<span class='notice'>You start entering an extraction signal to your handlers on [U]..</span>")
+	if(do_after(M, EXTRACTION_PHASE_PREPARE, target = M))
+		if(!U?.Adjacent(M) || extraction_deadline > world.time)
+			return
+		var/obj/effect/contractor_flare/F = new(get_turf(M))
+		extraction_flare = F
+		extraction_deadline = world.time + extraction_cooldown
+		M.visible_message("<span class='notice'>[M] enters a mysterious code on [U] and pulls a black and gold flare from [M.p_their()] belongings before lighting it.</span>",\
+						  "<span class='notice'>You finish entering the signal on [U] and light an extraction flare, initiating the extraction process.</span>")
+		addtimer(CALLBACK(src, .proc/extraction_step_1, U, M, F), EXTRACTION_PHASE_STEP_1)
+		extraction_timer_handle = addtimer(CALLBACK(src, .proc/deadline_reached), portal_duration, TIMER_STOPPABLE)
+
+/**
+  * First step of the extraction process.
```

Comment:
First? But there is no second?.. This is mildly confusing, especially given that the first step is dropping a flare (which happens before this step).. Rename it to `open_extraction_portal` or something